astropath/bayesian.py

```diff
@@ -0,0 +1,190 @@
+"""Methods related to Bayesian association analysis"""
+
+import numpy as np
+
+import copy
+
+from astropy import units
+from astropy.coordinates import SkyCoord
+
+from IPython import embed
+
+
+def raw_prior_Oi(Pchance, Sigma_m, method, half_light=None):
+    """
+    Raw prior for a given set of Pchance values and/or n(m)
+
+    Args:
+        Pchance (float or np.ndarray):
+            Chance probability
+        Sigma_m (float or np.ndarray):
+            Number density of sources on the sky brighter than m
+        method (str):
+            inverse :: Assign inverse to Sigma_m
+            inverse1 :: Assign inverse to Sigma_m * half_light
+            inverse2 :: Assign inverse to Sigma_m * half_light**2
+            orig_inverse :: Assign inverse to P_chance
+            identical :: All the same
+            linear :: 1-Pchance (not recommended)
+        half_light (float or np.ndarray, optional):
+            Angular size of the galaxy
+            Only required for several methods
+
+    Returns:
+        float or np.ndarray:
+
+    """
+    if method == 'linear':
+        return 1 - Pchance
+    elif method == 'orig_inverse':
+        return 1. / Pchance
+    elif method == 'inverse':
+        return 1. / Sigma_m
+    elif method == 'inverse1':
+        return 1. / Sigma_m / half_light
+    elif method == 'inverse2':
+        return 1. / Sigma_m / half_light**2
+    elif method == 'identical':
+        return np.ones_like(Pchance)
+    else:
+        raise IOError("Bad method {} for prior_Oi".format(method))
+
+
+def pw_Oi(r_w, theta_half, theta_prior, scale_half=1.):
+    """
+    Calculate p(w|O_i) for a given galaxy
+
+    Args:
+        r_w (np.ndarray):
+            offset from galaxy center in arcsec
+        theta_half (float):
+            Half-light radius of this galaxy in arcsec
+        theta_prior (dict):
+            Parameters for theta prior
+        scale_half (float):
```

Comment:
Please add some description for this input.

Reply:
Can also mention that only three methods: uniform, core and exp are supported for theta['method']

Reply:
done.  And I've added to the docs quite a bit on this.